### PR TITLE
place surface type const and unassemble flag in vtx and document effects

### PR DESCRIPTION
**Pull request by coco875**

### CODE REVIEW

File: courses/all_course_model.h [hunk 1]
```diff
@@ -10,26 +10,26 @@
...
-extern mk64_Vtx d_course_moo_moo_farm_vertex[0x1F24];
-extern mk64_Vtx d_course_banshee_boardwalk_vertex[0x1351];
-extern mk64_Vtx d_course_dks_jungle_parkway_vertex[0x162F];
-extern mk64_Vtx d_course_rainbow_road_vertex[0xC27];
-extern mk64_Vtx d_course_yoshi_valley_vertex[0xE88];
-extern mk64_Vtx d_course_bowsers_castle_vertex[0x2537];
-extern mk64_Vtx d_course_royal_raceway_vertex[0x2072];
-extern mk64_Vtx d_course_sherbet_land_vertex[0xA76];
-extern mk64_Vtx d_course_wario_stadium_vertex[0x17B3];
+extern Course_Vtx d_course_big_donut_vertex[0x48D];
```
Comment (MegaMech) [inline courses/all_course_model.h:13]: Can we do CourseVtx instead? Struct should be CamelCase with no underlines. We were just breaking the rules with the underline.

File: include/common_structs.h [hunk 2]
```diff
@@ -65,28 +65,32 @@ typedef enum {
...
+    /* -0x1 */ SURFACE_DEFAULT = -1,
     /* 0x00 */ AIRBORNE,
-    /* 0x01 */ PAVEMENT, // Luigi's Raceway, Toad's Turnpike, Koopa Troop beach shortcut tunnel, Mario Raceway, Royal Raceway, Rainbow Road, Block Fort, Double Deck, Skyscraper
-    /* 0x02 */ SAND_ONE, // Luigi's Raceway, Moo Moo Farm, Kalimiari Desert on course, Choco Mountain, Wario Stadium, DK Jungle on course, Yoshi Valley
-    /* 0x03 */ SAND_TWO, // Koopa Troopa Beach light color, Royal Raceway
-    /* 0x04 */ STONE_TWO, // Royal Raceway castle entrance, Bowser's Castle
-    /* 0x05 */ SNOW_ONE, // Frappe Snowland on course, Sherber Land tunnel
-    /* 0x06 */ STONE_ONE, // Royal Raceway castle bridges (even the wooden one), Banshee's Boardwalk, Big Donut
-    /* 0x07 */ SAND_FIVE, // Mario Raceway
+    /* 0x01 */ BITUMEN, // Luigi's Raceway, Toad's Turnpike, Koopa Troop beach shortcut tunnel, Mario Raceway, Royal Raceway, Rainbow Road, Block Fort, Double Deck, Skyscraper
```
Comment (MegaMech) [inline include/common_structs.h:74]: Asphalt?

File: include/common_structs.h [hunk 3]
```diff
@@ -65,28 +65,32 @@ typedef enum {
...
+    /* 0x02 */ DIRT, // Luigi's Raceway, Moo Moo Farm, Kalimiari Desert on course, Choco Mountain, Wario Stadium, DK Jungle on course, Yoshi Valley
+    /* 0x03 */ SAND, // Koopa Troopa Beach light color, Royal Raceway
+    /* 0x04 */ STONE, // Royal Raceway castle entrance, Bowser's Castle
+    /* 0x05 */ SNOW, // Frappe Snowland on course, Sherber Land tunnel
+    /* 0x06 */ BRIDGE, // Royal Raceway castle bridges (even the wooden one), Banshee's Boardwalk, Big Donut
+    /* 0x07 */ SAND_OFF_ROAD, // Mario Raceway
```
Comment (MegaMech) [inline include/common_structs.h:80]: Consider `SAND_OFFROAD`

File: include/common_structs.h [hunk 4]
```diff
@@ -65,28 +65,32 @@ typedef enum {
...
-    /* 0x10 */ WOOD_TWO, // Bowser's Castle bridge 2, DK Jungle bridge
-    /* 0x11 */ WOOD_ONE, // Frappe Snowland bridge, Bowser's Castle bridge 1,3, Yoshi Valley bridge 2
-    /* 0xFC */ BOOST_PAD_TWO = 0xFC, // DK Jungle
+    /* 0x0F */ CAVE, // DK Jungle cave
+    /* 0x10 */ ROPE_BRIDGE, // Bowser's Castle bridge 2, DK Jungle bridge
+    /* 0x11 */ WOOD_BRIDGE, // Frappe Snowland bridge, Bowser's Castle bridge 1,3, Yoshi Valley bridge 2
+    /* 0xFC */ BOOST_RAMP = 0xFC, // DK Jungle
     /* 0xFD */ OUT_OF_BOUNDS, // DK Jungle river island
-    /* 0xFE */ BOOST_PAD_ONE, // Royal Raceway
+    /* 0xFE */ BOOST_PAD, // Royal Raceway
```
Comment (MegaMech) [inline include/common_structs.h:93]: How do you feel about `GRAVITY_RAMP`? Cause it gives you slightly less gravity

**Review by MegaMech — CHANGES_REQUESTED** [review]

File: include/common_structs.h (hunk 2 above)
Comment (coco875) [reply include/common_structs.h:74]: yeah

File: include/common_structs.h (hunk 4 above)
Comment (coco875) [reply include/common_structs.h:93]: I think some ambiguous can be made with gravity where in game we don't have 0-gravity or not so much related to gravity it's a very small details and people we see it the first time can understand something totally different

Comment (MegaMech) [reply include/common_structs.h:93]: Hmm... Not quite a fan of having two differing terms that mean the same thing: boost_ramp and boost_pad.
How about boost_ramp2 or boost_ramp_fly or _float or something?

File: src/racing/memory.c [hunk 5]
```diff
@@ -430,32 +430,32 @@ void func_802A86A8(Course_Vtx *data, u32 arg1) {
...
 
         flags = temp_a0 & 3;
         flags |= (temp_a3 << 2) & 0xC;
 
-        vtx->v.ob[2] = course_vtx->ob[2];
-        vtx->v.tc[0] = course_vtx->tc[0];
-        vtx->v.tc[1] = course_vtx->tc[1];
+        vtx->v.ob[2] = CourseVtx->ob[2];
+        vtx->v.tc[0] = CourseVtx->tc[0];
+        vtx->v.tc[1] = CourseVtx->tc[1];
```
Comment (MegaMech) [inline src/racing/memory.c:447]: courseVtx for function variables

**Review by MegaMech — CHANGES_REQUESTED** [review]

File: src/racing/memory.c (hunk 5 above)
Comment (coco875) [reply src/racing/memory.c]: fix

**Review by MegaMech — APPROVED** [review]

**Pull request merged**